zipline/pipeline/factors/factor.py

```diff
@@ -1205,17 +1205,22 @@ def compute(self, today, assets, out, close):
     '''
     dtype = float64_dtype
 
-    def __getattr__(self, attribute_name):
+    def __getattribute__(self, name):
+        outputs = object.__getattribute__(self, 'outputs')
+        if outputs is not NotSpecified and name in outputs:
+            return RecarrayField(factor=self, attribute=name)
+        else:
+            return super(CustomFactor, self).__getattribute__(name)
+
+    def __getattr__(self, name):
```

Comment:
I would probably just remove this method and make the logic in `__getattribute__` look like:

```
if outputs is NotSpecified:
    return super(...).__getattribute__(...)
if name in outputs:
    return RecarrayField(...)
raise AttributeError(...)
```